### PR TITLE
Feature/track sent requests (close #69 and #68)

### LINKED ISSUE #68: Sent requests UI

### LINKED ISSUE #69: Query the DB for requests a user sent

### PR DESCRIPTION
**Pull request by axelmarmet**
close #69 and close #68

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/balelecbud/SocialActivityTest.java [hunk 1]
```diff
@@ -81,10 +82,10 @@ private void createFriendship(User user) {
         mockDb.storeDocumentWithID(DatabaseWrapper.FRIENDSHIPS_PATH, user.getUid(), toStore);
     }
 
-    private void createRequestFromUser(User user) {
+    private void createRequest(User from, User to) {
```
Comment (Pialex99) [inline app/src/androidTest/java/ch/epfl/balelecbud/SocialActivityTest.java:85]: You could replace `FriendshipUtils.addFriend(User)` with this function as they are similar

File: app/src/androidTest/java/ch/epfl/balelecbud/util/database/MockDatabaseWrapper.java [hunk 2]
```diff
@@ -162,16 +166,64 @@ private Field getField(MyQuery.WhereClause clause, Class clazz) {
...
+                return friendships;
+            default:
+                throw new IllegalArgumentException("Unsupported collection name " + name);
+        }
+    }
+
+    private List<Map<String, Boolean>> filterMapValues(Map<String, Map<String, Boolean>> mapToFilter, List<MyQuery.WhereClause> clauses) {
+        List<Map<String, Boolean>> values = new LinkedList<>(mapToFilter.values());
+        for (MyQuery.WhereClause clause : clauses) {
+            List<Map<String, Boolean>> newValues = new LinkedList<>();
```
Comment (Pialex99) [inline app/src/androidTest/java/ch/epfl/balelecbud/util/database/MockDatabaseWrapper.java:206]: You could use
`values = values.stream().filter(oldValue -> oldValue.getOrDefault(clause.getLeftOperand(), false)).collect(Collectors.toList());`

File: app/src/main/java/ch/epfl/balelecbud/util/database/MyQuery.java [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
 
+import ch.epfl.balelecbud.util.StringUtils;
+
 public class MyQuery {
 
     private final String collectionName;
     private final List<WhereClause> whereClauses;
 
+    public MyQuery(String collectionName, WhereClause whereClause){
+        this(collectionName, Lists.newArrayList(whereClause));
```
Comment (Pialex99) [inline app/src/main/java/ch/epfl/balelecbud/util/database/MyQuery.java:17]: You could use `Collections.singletonList(whereClause)` but it also works fine

File: app/src/androidTest/java/ch/epfl/balelecbud/util/database/MockDatabaseWrapper.java (hunk 2 above)
Comment (axelmarmet) [reply app/src/androidTest/java/ch/epfl/balelecbud/util/database/MockDatabaseWrapper.java:206]: I personally agree and like this functional style but not sure about the rest of the team

**Pull request merged**